Lib/distutils/dist.py

```diff
@@ -597,11 +600,26 @@ def _parse_command_opts(self, parser, args):
             if help_option_found:
                 return
 
+        global_option_names = {n for n, *_ in self.global_options}
+        command_option_names = {n for n, *_ in cmd_class.user_options}
+
         # Put the options from the command-line into their official
         # holding pen, the 'command_options' dictionary.
         opt_dict = self.get_option_dict(command)
         for (name, value) in vars(opts).items():
-            opt_dict[name] = ("command line", value)
+            opt_name = name.translate(longopt_revxlate)
```

Comment:
It would be good to add a comment explaining why you need to replace the underscores with hyphens.

Why not just use `name.replace`? Is there a better reason to use `str.maketrans` instead?

Reply:
The main reason is because I wanted to make it a little bit more explicit (without success) that this is the "reverse" `longopt_xlate` translation that exists in `fancy_getopt`.

I don't think there's any other reason to use `translate` instead of `replace`. I will follow your suggestion and use `replace` instead.